app/scripts/modules/core/securityGroup/filter/SecurityGroupFilterCtrl.ts

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+import {compact, uniq, map} from 'lodash';
+import {module} from 'angular';
+import {Application} from '../../application/application.model';
+
+export const SECURITY_GROUP_FILTER = 'securityGroup.filter.controller';
+
+let ngmodule = module(SECURITY_GROUP_FILTER, [
+  require('./securityGroup.filter.service.js'),
+  require('./securityGroup.filter.model.js'),
+  require('../../filterModel/dependentFilter/dependentFilter.service.js'),
+  require('./securityGroupDependentFilterHelper.service.js'),
+]);
+
+export class SecurityGroupFilterCtrl {
+  private app: Application;
+  public regionHeadings: any;
+  public accountHeadings: any;
+  public tags: any[];
+  public stackHeadings: string[];
+  public providerTypeHeadings: any[];
+
+  static get $inject() {
+    return [
+      'securityGroupFilterService',
+      'SecurityGroupFilterModel',
+      'dependentFilterService',
+      'securityGroupDependentFilterHelper',
+      '$scope',
+      '$rootScope',
+    ];
+  }
+
+  constructor(private securityGroupFilterService: any,
+              private SecurityGroupFilterModel: any,
+              private dependentFilterService: any,
+              private securityGroupDependentFilterHelper: any,
+              private $scope: angular.IScope,
+              private $rootScope: angular.IScope,
+  ) {
+  }
+
+  public $onInit() {
+    let { $scope, $rootScope, app, SecurityGroupFilterModel, securityGroupFilterService } = this;
+
+    $scope.sortFilter = SecurityGroupFilterModel.sortFilter;
+    this.tags = SecurityGroupFilterModel.tags;
+
+    $scope.$on('$destroy', $rootScope.$on('$locationChangeSuccess', () => {
+      SecurityGroupFilterModel.activate();
+      securityGroupFilterService.updateSecurityGroups(app);
+    }));
+
+    this.initialize();
+    app.securityGroups.onRefresh($scope, () => this.initialize());
+  }
+
+  private updateSecurityGroups() {
+    let {
```

Comment:
note there's a bunch of ES6 that's using `let { foo ... } = ` destructuring.  I'll just leave those as is and update only the PR